content/blog/2025-04-10-fastest-tpch-generator.md

```diff
@@ -0,0 +1,617 @@
+or CSV, tpchgen-cli creates the full SF=100 parquet format dataset in less than
+45 seconds.
+
+[added support for CSV]: https://github.com/clflushopt/tpchgen-rs/pull/54
+[@niebayes]: https://github.com/niebayes
+[fixing]: https://github.com/clflushopt/tpchgen-rs/pull/66 
+[several]: https://github.com/clflushopt/tpchgen-rs/issues/73 
+[bugs]: https://github.com/clflushopt/tpchgen-rs/issues/65
+[made]: https://github.com/clflushopt/tpchgen-rs/pull/71
+[tpchgen-arrow]: https://crates.io/crates/tpchgen-arrow
+[Apache Arrow]: https://arrow.apache.org/
+[a small 300 line PR]: https://github.com/clflushopt/tpchgen-rs/pull/61
+[Apache Arrow](https://arrow.apache.org/
+[a small 300 line PR]: https://github.com/clflushopt/tpchgen-rs/pull/61
+[Rust Parquet writer]: https://crates.io/crates/parquet
+
+<table>
+  <tr>
+   <td>Scale Factor
+   </td>
+   <td>Time to generate Parquet
+   </td>
+   <td>Speed compared to tbl generation
+   </td>
+  </tr>
+  <tr>
+   <td>1
+   </td>
+   <td>0m1.649s
+   </td>
+   <td>0.8
+   </td>
+  </tr>
+  <tr>
+   <td>10
+   </td>
+   <td>0m5.643s
+   </td>
+   <td>0.7
+   </td>
+  </tr>
+  <tr>
+   <td>100
+   </td>
+   <td>0m45.243s
+   </td>
+   <td>0.7
+   </td>
+  </tr>
+  <tr>
+   <td>100 (to /dev/null)
+   </td>
+   <td>0m45.153s
+   </td>
+   <td>0.5
+   </td>
+  </tr>
+</table>
+
+
+**Table 5**: `tpchgen-cli` Parquet generation performance measured with  `time
+target/release/tpchgen-cli -s $SCALE_FACTOR --format=parquet`
+
+
+# Conclusion 👊🎤
+
+With a few days, some fellow database nerds, and the power of Rust we made
+something 10x better than currently exists. We hope it inspires more research
+into analytical systems using the TPCH dataset and that people build awesome
+things with it. For example, Sean has already added [on-demand generation of
+tables to GlareDB]. Please consider joining us and helping out at
+[https://github.com/clflushopt/tpchgen-rs].
+
+We met while working together on Apache DataFusion in various capacities. If you
+are looking for a community of like minded people hacking on databases, we
+welcome you to [come join us]. We are in the process of integrating this into
+DataFusion (see [apache/datafusion#14608]) if you are interested in helping 🎣
+
+[on-demand generation of tables to GlareDB]: https://github.com/GlareDB/glaredb/pull/3549
+[https://github.com/clflushopt/tpchgen-rs]: https://github.com/clflushopt/tpchgen-rs
+[come join us]: https://datafusion.apache.org/contributor-guide/communication.html
+[apache/datafusion#14608]: https://github.com/apache/datafusion/issues/14608
+
+# About the Authors:
+
+- [Andrew Lamb] ([@alamb]) is a Staff Engineer at [InfluxData] and a PMC member of [Apache DataFusion] and [Apache Arrow].
+- Achraf B ([@clflushopt]) is a Software Engineer at [Optable] where he works on data infrastructure.
+- [Sean Smith] ([@scsmithr]) is the founder of  focused on building a fast analytics database.
```

Comment:
```suggestion
- [Sean Smith] ([@scsmithr]) is the founder of [GlareDB] focused on building a fast analytics database.
```

Reply:
Correct me if I'm wrong, based on the LinkedIn link :-)

Reply:
Yes for sure -- great catch